projectFile.js

```diff
@@ -14,16 +15,34 @@ function openProjectRequest(parent) {
 }
 
 function saveProjectRequest(parent) {
-  let filename = dialog.showSaveDialogSync(parent, {
-    properties: ['showOverwriteConfirmation'],
-    title: 'Save project file...',
-    filters: [{ name: 'Project file (*.rpe)', extensions: ['rpe'] }],
-    defaultPath: 'project.rpe',
+  const folderPath = dialog.showSaveDialogSync(parent, {
+    title: 'Save project folder...',
+    defaultPath: 'project',
+    buttonLabel: 'Save Project Folder',
+    properties: ['createDirectory'],
   });
-  if (filename !== undefined) {
-    if (path.extname(filename) !== '.rpe') filename += '.rpe';
+
+  if (folderPath) {
+    if (!fs.existsSync(folderPath)) {
+      fs.mkdirSync(folderPath, { recursive: true });
+    }
+
+    const folderName = path.basename(folderPath);
+    const projectFileName = `${folderName}.rpe`;
+    const projectFilePath = path.join(folderPath, projectFileName);
+    fs.writeFileSync(projectFilePath, '');
+
+    const currentWindow = BrowserWindow.getFocusedWindow();
+    if (currentWindow) {
+      currentWindow.setTitle(`${projectFileName} - Rapid Power Estimator`);
+    } else {
+      console.error('No focused window found to set title.');
+    }
+
+    return projectFilePath;
   }
-  return filename === undefined ? '' : filename;
+
+  return '';
```

Comment:
In the earlier approach, the code checked if the file name was undefined before returning an empty string, now it just simplifies things by directly returning an empty string, since the goal is to return an empty string when no file is selected, I believe there is no need for extra checks (undefined), returning the empty string does the same job.